### PR TITLE
#169817895 Pre Screened Supplier can create Accommodation

### PR DESCRIPTION
**Pull request by Jajabenit250**
#### What does this PR do?
- Pre Screened Supplier should be able to create Accommodation facilities to barefoot Nomad
#### Description of Task to be completed?
- Pre Screened Supplier should be able to create Accommodation facilities 
- Check if user is loggedIN as Supplier
- return to the user the Requests Data (created Accommodation)
#### How should this be manually tested?
-  Clone this repository
-  git checkout feature/169817895/prescreened-supplier-can-create-accomodation-facility
-  open console and type and run(- npm install) to install all dependencies
-  and then run (- npm start) to start the server
#### Any background context you want to provide?
- User need to be logged in before Creating Accommodation
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/169817895
#### Screenshots (if appropriate) 
ezgif com-video-to-gif (1)

#### Questions:

### CODE REVIEW

File: src/controllers/accomodation.controller.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      response.errorMessage(res, 'You are not authorized to perform this action', 401, 'error');
+    }
+    const newAccomodation = {
+      name: accommodationName, description, locationId, category, owner
+    };
+    const createdAccommodation = await accomodationServices.createAccomodation(newAccomodation);
+    const accomodationId = createdAccommodation.dataValues.id;
+    const accommodationImages = await AccommodationHelper.saveImages(images, accomodationId);
+    const accommodationRooms = await AccommodationHelper.createRooms(rooms, accomodationId);
+    const accommodationServices = await AccommodationHelper.createAccommodationServices(services, accomodationId);
```
Comment (mujohn26) [inline src/controllers/accomodation.controller.js]: Good work benit but what does this function `createAccommodationServices` do ?,is it creating accommodation services ?,surely am not getting what it is doing specifically.

Comment (Jajabenit250) [reply src/controllers/accomodation.controller.js]: An accommodation might be able to provide other services that they provide. like restaurant, club and bars, etc .... @mujohn26 that's why i did it.

**Review by mujohn26 — APPROVED** [review]

File: src/helpers/accomodation.helper.js [hunk 2]
```diff
@@ -30,6 +32,86 @@ class Accomodation {
...
+        createdRooms,
+        roomTypeImage
+      };
+      return data;
+    }));
+  }
+
+  /**
+   * This function helps to create services of accommodation
+   * @param { Array } services images of accomodation
```
Comment (jo-rdan) [inline src/helpers/accomodation.helper.js]: @Jajabenit250 update the description on this param it is a bit misleading.

File: src/helpers/accomodation.helper.js [hunk 3]
```diff
@@ -30,6 +32,86 @@ class Accomodation {
...
+      const newServices = { name: serviceName, accomodationId };
+      const createdServices = await accomodationServices.createAccommodationService(newServices);
+      const data = createdServices.dataValues;
+      return data;
+    }));
+  }
+
+  /**
+   * This function helps to create amenities of accommodation
+   * @param { Array } amenities images of accomodation
```
Comment (jo-rdan) [inline src/helpers/accomodation.helper.js]: Update this line too, the JS-doc should be as more descriptive as possible

**Review by jo-rdan — CHANGES_REQUESTED** [review]
@Jajabenit250 good work done really! however I left few comments on your PR update them and we're good to go, and try to update your JS-docs to be very descriptive because we will be using those. thank you!

**Review by jo-rdan — APPROVED** [review]

**Review by bdushimi — APPROVED** [review]

**Pull request merged**